core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala

```diff
@@ -131,6 +140,54 @@ class RawTxFinalizerTest extends BitcoinSAsyncTest {
     testBIP69Finalizer(sortedInputs, sortedOutputs)
   }
 
+  it must "shuffle inputs" in {
+    forAllAsync(CreditingTxGen.inputsAndOutputs()) {
+      case (inputs, outputs) =>
+        val txBuilder = RawTxBuilder() ++= inputs ++= outputs
+        val finalized = txBuilder.setFinalizer(ShuffleInputsFinalizer)
+        val txsF =
+          FutureUtil.foldLeftAsync(Vector.empty[Transaction], 0 to 20) {
+            (accum, _) => finalized.buildTx().map(_ +: accum)
+          }
+        txsF.map(txs =>
+          assert(
+            inputs.size <= 1 || txs.exists(
+              _.inputs.map(_.previousOutput) != inputs.map(_.outPoint))))
```

Comment:
If the number of inputs is small, won't this assertion have a decent chance of failing?

Reply:
that's why we generate 20 of them